dpnp/dpnp_iface_sorting.py

```diff
@@ -190,18 +246,34 @@ def sort(x1, **kwargs):
     Examples
     --------
     >>> import dpnp as np
-    >>> a = np.array([1, 4, 3, 1])
-    >>> out = np.sort(a)
-    >>> [i for i in out]
-    [1, 1, 3, 4]
+    >>> a = np.array([[1,4],[3,1]])
+    >>> np.sort(a)                # sort along the last axis
+    array([[1, 4],
+           [1, 3]])
+    >>> np.sort(a, axis=None)     # sort the flattened array
+    array([1, 1, 3, 4])
+    >>> np.sort(a, axis=0)        # sort along the first axis
+    array([[1, 1],
+           [3, 4]])
 
     """
 
-    x1_desc = dpnp.get_dpnp_descriptor(x1, copy_when_nondefault_queue=False)
-    if x1_desc and not kwargs:
-        if x1_desc.ndim != 1:
-            pass
-        else:
-            return dpnp_sort(x1_desc).get_pyobj()
-
-    return call_origin(numpy.sort, x1, **kwargs)
+    if kind is not None:
+        pass
+    elif order is not None:
+        pass
+    else:
+        dpnp.check_supported_arrays_type(a)
+        if a.ndim == 0:
+            raise numpy.AxisError(
```

Comment:
@antonwolfy 
For zero dimensional array, `NumPy` and `dpctl` behave differently.
`dpctl` is consistent for both `sort` and `argsort` but `NumPy` is not.
Although `dpctl` approach might be better since it is consistent between the two functions, I followed `NumPy` behavior to make sure all cupy tests will pass.
 
`>>> a=numpy.array(2.5)`
`>>> numpy.sort(a)`
`AxisError: axis -1 is out of bounds for array of dimension 0`
`>>> numpy.argsort(a)`
`array([0])`
`>>> b=dpt.asarray(2.5)`
`>>> dpt.sort(b)`
`usm_ndarray(2.5, dtype=float32)`
`>>> dpt.argsort(b)`
`usm_ndarray(0)`